admin/server/deployment.go

```diff
@@ -309,16 +306,34 @@ func (s *Server) CreateDeployment(ctx context.Context, req *adminv1.CreateDeploy
 	var slots int
 	switch req.Environment {
 	case "prod":
+		if req.Branch != "" {
+			return nil, status.Error(codes.InvalidArgument, "branch cannot be specified for prod deployments")
+		}
+		if req.Editable {
+			return nil, status.Error(codes.InvalidArgument, "editable cannot be set for prod deployments")
+		}
 		branch = ""
 		slots = proj.ProdSlots
 	case "dev":
-		// Generate a random branch name for dev deployments
-		b := make([]byte, 8)
-		_, err := rand.Read(b)
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
+		if req.Branch != "" {
+			branch = req.Branch
+			// only one deployment per branch allowed
+			depl, err := s.admin.DB.FindDeploymentsForProject(ctx, proj.ID, "dev", branch)
+			if err != nil && !errors.Is(err, database.ErrNotFound) {
+				return nil, status.Error(codes.InvalidArgument, err.Error())
+			}
```

Comment:
`database.ErrNotFound` doesn't need to be handled here, it can't be returned by database calls that returns lists